### DISCUSSION TITLE
Implementing a Navwalker Mega menu?

### DISCUSSION THREAD

**Discussion by wpbuilds** [understrap/understrap-child#362] [Q&A]
Hi there, just wondering if anyone has attempted to build/integrate a Navwalker Mega Menu into the child-theme? (No plugins) Having never developed a custom Navwalker menu before, I was looking for an up-to-date repo/tutorial for Bootstrap 5 implementation - as a starting point.

Any help, guidance, link to repo would be gratefully appreciated :-)

**Comment by bacoords** [comment]
Interesting - we've done some things with ACF- [adding custom fields to the default WP menu](https://www.advancedcustomfields.com/resources/adding-fields-menu-items/) and then hooking into the navwalker to build out content. We've also just skipped wp menus entirely and used an [ACF Options page](https://www.advancedcustomfields.com/resources/acf_add_options_page/) to help us build one.

Do you have an example of a "mega menu" built with Bootstrap that you're trying to replicate in WP?

**Reply by wpbuilds** [reply]
Hi @bacoords and thanks for the info :-) - an interesting approach using ACF, I had not considered that? A future method to consider! In the end I found a couple of old (4-5 years) BS4 repos that used a custom Navwalker and WP widgets to serve up content in a Mega Menu. Neither of the repos initially worked out of the box with the latest understrap child theme, and accessibility seemed to be a secondary consideration. But after much reading and adjustment of classes I cobbled together a working solution for BS5. 

Although, I am not a great fan of the Mega Menu my clients seem to regularly ask for them. It would certainly be a "nice to have" option built into the theme?

**Reply by bacoords** [reply]
Sure - I would just be curious what a Bootstrap-based mega menu would look like, if you have any examples. Typically we end up building mega menus from scratch or using a plugin, I've never actually built one using strictly Bootstrap elements before.

**Reply by wpbuilds** [reply]
Yeah, I have also used plugins before too but you can end up with bloat and unnecessary functionality. 

Here's the link to the old repo I used as a starting point -  https://github.com/jaycbrf4/WP-Bootstrap4-MegaMenu-Navwalker.

Once I got it working I liked the fact that it dynamically created sidebar (widgets) based on a single class you add to a WP menu item. Obvs there is a fair bit of styling required for the the menu items and widgets you wanna use - worked for my needs. Could post a pic when I'm done if ya wanna see what it looks like?

**Reply by bacoords** [reply]
Woah that's pretty cool.  I'm very curious what it actually looks like on the front end.

**Reply by wpbuilds** [reply]
When I've finished the build (nov-dec) I'll post a link to see it.

**Reply by wpbuilds** [reply]
@bacoords I totally forgot to post my Mega Menu implementation for you to see... doh! Here it is - [https://nomensa.com](url)

**Reply by bacoords** [reply]
🎉 That's awesome!
